pkg/node/manager.go

```diff
@@ -109,3 +137,120 @@ func DeleteNode(ni Identity) {
 
 	mutex.Unlock()
 }
+
+// updateIPRoute updates the IP routing entry for the given node n via the
+// network interface that as ownAddr.
+func updateIPRoute(ni Identity, oldNode, n *Node, ownAddr net.IP) {
+	nodeIPv6 := n.GetNodeIP(true)
+	log.Debugf("iproute: Setting endpoint v6 route for %s via %s",
+		n.IPv6AllocCIDR, nodeIPv6)
+
+	nl, err := firstLinkWithv6(ownAddr)
+	if err != nil {
+		log.WithError(err).Errorf("iproute: Unable to get v6 interface with IP %s", ownAddr)
+		return
+	}
+	dev := nl.Attrs().Name
+	if dev == "" {
+		log.Errorf("iproute: Unable to get v6 interface for %s: empty interface name", ownAddr)
+		return
+	}
+
+	if oldNode != nil {
+		oldNodeIPv6 := oldNode.GetNodeIP(true)
+		if oldNode.IPv6AllocCIDR.String() != n.IPv6AllocCIDR.String() ||
+			!oldNodeIPv6.Equal(nodeIPv6) ||
+			oldNode.dev != n.dev {
+			// If any of the routing components changed, then remove the old entries
+
+			err = routeDel(oldNodeIPv6.String(), oldNode.IPv6AllocCIDR.String(), oldNode.dev)
+			if err != nil {
+				log.Warning(err)
+			}
+		}
+	} else {
+		n.dev = dev
+	}
+
+	// Always re add
+	err = routeAdd(nodeIPv6.String(), n.IPv6AllocCIDR.String(), dev)
+	if err != nil {
+		log.Warning(err)
+		return
+	}
+}
+
+// deleteIPRoute deletes the routing entries previously created for the given
+// node.
+func deleteIPRoute(node *Node) {
+	oldNodeIPv6 := node.GetNodeIP(true)
+
+	err := routeDel(oldNodeIPv6.String(), node.IPv6AllocCIDR.String(), node.dev)
+	if err != nil {
+		log.Warning(err)
+	}
+}
+
+// firstLinkWithv6 returns the first network interface that contains the given
+// IPv6 address.
+func firstLinkWithv6(ip net.IP) (netlink.Link, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, err
+	}
+	for _, l := range links {
+		addrs, _ := netlink.AddrList(l, netlink.FAMILY_V6)
+		for _, a := range addrs {
+			if ip.Equal(a.IP) {
+				return l, nil
+			}
+		}
+	}
+
+	return nil, fmt.Errorf("No address found")
+}
+
+func routeAdd(dstNode, podCIDR, dev string) error {
+	prog := "ip"
+
+	// for example: ip -6 r a fd00::b dev eth0
+	// TODO: don't add direct route if a subnet of that IP is already present
+	// in the routing table
+	args := []string{"-6", "route", "add", dstNode, "dev", dev}
+	out, err := exec.Command(prog, args...).CombinedOutput()
```

Comment:
Would it be feasible to use `runProg` from here: https://github.com/cilium/cilium/blob/master/daemon/daemon.go#L389 ?

Reply:
It would, but that function needs to be else were, maybe under `common/utils`